src/main.cpp

```diff
@@ -87,6 +87,7 @@ namespace
 			return;
 		}
 
+		static size_t current_renderer_index = 0; // 
```

Comment:
We don't need to track this in a persistent variable at all, since it can be trivially calculated using pointer arithmetic against the value of `renderers::all().data()`:

```cpp
const auto current_renderer_index = static_cast<size_t>(regular_renderer.description - renderers::all().data());
```

Just do the above when you need it. The less state a program has, the better 😄